isic/core/api.py

```diff
@@ -57,20 +68,17 @@ class ImageViewSet(ReadOnlyModelViewSet):
     )
     @action(detail=False, methods=['get'], pagination_class=None)
     def facets(self, request):
-        serializer = SearchQuerySerializer(data=request.query_params)
-        serializer.is_valid(raise_exception=True)
-        query = build_filtered_query(request.user, serializer.data.get('query'))
-        return Response(facets(query))
+        query = build_filtered_query(request.user, request.query_params)
+        collection_pks = query['bool'].get('filter', {}).get('terms', {}).get('collections')
```

Comment:
Can this be moved inside `search.facets`, since it's based on `query` anyway? Also, a hefty code comment explaining that this is to prevent the identifiers (PKs) of other facets from being included (with counts of 0) in the facet output.